### PR TITLE
fix: adjust adaptiveCard/action handling

### PR DESCRIPTION
**Pull request by mdrichardson**
#minor

## Description
Per [the OBI](https://github.com/microsoft/botframework-obi/blob/jeffders/adaptiveCards2/protocols/botframework-protocol/botframework-action-execute.md#authentication-field), `AdaptiveCardInvokeValue.authentication` should actually be a `TokenExchangeInvokeRequest`, which has the same properties as `AdaptiveCardAuthentication`, which is now superfluous.

In working on this, I also noticed that [this PR](https://github.com/microsoft/botbuilder-js/pull/3716) was actually already covered.

## Specific Changes

  - Remove `AdaptiveCardAuthentication` and replace with `TokenExchangeInvokeRequest`
  - Revert [this PR](https://github.com/microsoft/botbuilder-js/pull/3716) (manually, because I didn't realize I should've just called `git revert` until I had already manually removed it.
  - Added tests

### CODE REVIEW

File: libraries/botbuilder/src/teamsActivityHandler.ts [hunk 1]
```diff
@@ -133,9 +133,6 @@ export class TeamsActivityHandler extends ActivityHandler {
                             await this.handleTeamsTabSubmit(context, context.activity.value)
                         );
 
-                    case 'adaptiveCard/action':
```
Comment (mdrichardson) [inline libraries/botbuilder/src/teamsActivityHandler.ts:136]: Note: Everything removed in this file and related tests is already covered on new line 138, when the `default` of the switch/case calls `super.onInvokeActivity()`

**Review by joshgummersall — APPROVED** [review]
@mdrichardson was the initial PR that introduced this functionality necessary in the first place? It seems like the handling code already existed.

**Comment by mdrichardson** [comment]
I don't believe it was necessary, since yes, since it was handled more broadly [here](https://github.com/microsoft/botbuilder-js/pull/3327)

**Comment by joshgummersall** [comment]
@cfe84, if you have a moment, did you try to leverage the action in question before submitting https://github.com/microsoft/botbuilder-js/pull/3716?

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**